rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -508,3 +509,25 @@ TEST_F(TestExecutor, is_spinning) {
 
   ASSERT_TRUE(timer_called);
 }
+
+TEST_F(TestExecutor, release_ownership_entity_after_spinning_cancel) {
+  using namespace std::chrono_literals;
+
+  // Create an Executor
+  rclcpp::executors::SingleThreadedExecutor executor;
+
+  auto future = std::async(std::launch::async, [&executor] {executor.spin();});
+
+  auto node = std::make_shared<rclcpp::Node>("test_node");
+  auto callback = [](
+    const test_msgs::srv::Empty::Request::SharedPtr, test_msgs::srv::Empty::Response::SharedPtr) {
+    };
+  auto server = node->create_service<test_msgs::srv::Empty>("test_service", callback);
+  executor.add_node(node);
+  std::this_thread::sleep_for(50ms);
+  executor.cancel();
```

Comment:
this test is not robust.
it could happen that we call `executor.cancel();` before the executors starts to spinning.
This will result in the future timing out.

We should wait until the executor is spinning before cancelling it.
I would suggest the following:
```
while (!executor.is_spinning()) { std::this_thread::sleep_for(std::chrono::milliseconds(1); }
executor.add_node(node);
std::this_thread::sleep_for(50ms);
executor.cancel();
```

Reply:
Yes. is_spinning() should be used.